contracts/access/manager/AccessManager.sol

```diff
@@ -0,0 +1,703 @@
+     * an operation that is restricted to this group.
+     */
+    function getGroupAdmin(uint256 groupId) public view virtual returns (uint256) {
+        return _groups[groupId].admin;
+    }
+
+    /**
+     * @dev Get the group that acts as a guardian for a given group.
+     *
+     * The guardian permission allows canceling operations that have been scheduled under the group.
+     */
+    function getGroupGuardian(uint256 groupId) public view virtual returns (uint256) {
+        return _groups[groupId].guardian;
+    }
+
+    /**
+     * @dev Get the group current grant delay, that value may change at any point, without an event emitted, following
+     * a call to {setGrantDelay}. Changes to this value, including effect timepoint are notified by the
+     * {GroupGrantDelayChanged} event.
+     */
+    function getGroupGrantDelay(uint256 groupId) public view virtual returns (uint32) {
+        return _groups[groupId].delay.get();
+    }
+
+    /**
+     * @dev Get the access details for a given account in a given group. These details include the timepoint at which
+     * membership becomes active, and the delay applied to all operation by this user that require this permission
+     * level.
+     *
+     * Returns:
+     * [0] Timestamp at which the account membership becomes valid. 0 means role is not granted.
+     * [1] Current execution delay for the account.
+     * [2] Pending execution delay for the account.
+     * [3] Timestamp at which the pending execution delay will become active. 0 means no delay update is scheduled.
+     */
+    function getAccess(uint256 groupId, address account) public view virtual returns (uint48, uint32, uint32, uint48) {
+        Access storage access = _groups[groupId].members[account];
+
+        uint48 since = access.since;
+        (uint32 oldDelay, uint32 newDelay, uint48 effect) = access.delay.split();
+
+        return effect.isSetAndPast(Time.timestamp()) ? (since, newDelay, 0, 0) : (since, oldDelay, newDelay, effect);
+    }
+
+    /**
+     * @dev Check if a given account currently had the permission level corresponding to a given group. Note that this
+     * permission might be associated with a delay. {getAccess} can provide more details.
+     */
+    function hasGroup(uint256 groupId, address account) public view virtual returns (bool) {
+        if (groupId == PUBLIC_GROUP) {
+            return true;
+        } else {
+            (uint48 inGroupSince, , , ) = getAccess(groupId, account);
+            return inGroupSince.isSetAndPast(Time.timestamp());
+        }
+    }
+
+    // =============================================== GROUP MANAGEMENT ===============================================
+    /**
+     * @dev Give a label to a group, for improved group discoverabily by UIs.
+     *
+     * Emits a {GroupLabel} event.
+     */
+    function labelGroup(uint256 groupId, string calldata label) public virtual onlyGroup(ADMIN_GROUP) {
```

Comment:
I feel like we should use a single modifier for all the admin-restricted functions
```suggestion
    function labelGroup(uint256 groupId, string calldata label) public virtual onlyDelayedAdmin() {
```